cpp/src/Ice/Properties.cpp

```diff
@@ -2,8 +2,729 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-#include <Ice/Properties.h>
+#include "Ice/Properties.h"
+#include "IceUtil/StringUtil.h"
+#include "IceUtil/FileUtil.h"
+#include "Ice/Initialize.h"
+#include "Ice/LocalException.h"
+#include "Ice/PropertyNames.h"
+#include "Ice/Logger.h"
+#include "Ice/LoggerUtil.h"
 
-Ice::Properties::~Properties()
+#include <fstream>
+
+using namespace std;
+using namespace Ice;
+using namespace IceInternal;
+
+Ice::Properties::Properties(const Properties& p)
+{
+    lock_guard lock(const_cast<Properties&>(p)._mutex);
```

Comment:
Maybe we should just make the mutex mutable?

Reply:
Fixed as suggested.